tests/components/analytics/test_analytics.py

```diff
@@ -1426,3 +1440,264 @@ async def async_modify_analytics(
+
+    await analytics.send_snapshot()
+
+    await analytics.save_preferences({ATTR_BASE: False, ATTR_SNAPSHOTS: True})
+    await analytics.send_snapshot()
+
+    await analytics.save_preferences({ATTR_BASE: True, ATTR_SNAPSHOTS: False})
+    await analytics.send_snapshot()
+
+    assert len(aioclient_mock.mock_calls) == 0
+
+
+async def test_send_snapshot_success(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test successful snapshot submission."""
+    snapshot_endpoint = ANALYTICS_SNAPSHOT_ENDPOINT_URL
+    aioclient_mock.post(
+        snapshot_endpoint,
+        status=200,
+        json={"submission_identifier": "test-identifier-123"},
+    )
+    analytics = Analytics(hass)
+    await analytics.save_preferences({ATTR_BASE: True, ATTR_SNAPSHOTS: True})
+
+    await analytics.send_snapshot()
+
+    assert len(aioclient_mock.mock_calls) == 1
+    assert analytics._data.submission_identifier == "test-identifier-123"
+    assert "Submitted snapshot analytics to Home Assistant servers" in caplog.text
+
+
+async def test_send_snapshot_with_existing_identifier(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test snapshot submission with existing identifier."""
+    snapshot_endpoint = ANALYTICS_SNAPSHOT_ENDPOINT_URL
+    aioclient_mock.post(
+        snapshot_endpoint,
+        status=200,
+        json={"submission_identifier": "test-identifier-123"},
+    )
+    analytics = Analytics(hass)
+    analytics._data.submission_identifier = "old-identifier"
+    await analytics.save_preferences({ATTR_BASE: True, ATTR_SNAPSHOTS: True})
+
+    await analytics.send_snapshot()
+
+    assert len(aioclient_mock.mock_calls) == 1
+    call_headers = aioclient_mock.mock_calls[0][3]
+    assert call_headers["X-Device-Database-Submission-Identifier"] == "old-identifier"
+    assert analytics._data.submission_identifier == "test-identifier-123"
+    assert "Submitted snapshot analytics to Home Assistant servers" in caplog.text
+
+
+async def test_send_snapshot_invalid_identifier(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test snapshot submission with invalid identifier."""
+    snapshot_endpoint = ANALYTICS_SNAPSHOT_ENDPOINT_URL
+    aioclient_mock.post(
+        snapshot_endpoint,
+        status=400,
+        json={
+            "kind": "invalid-submission-identifier",
+            "message": "The identifier is invalid",
+        },
+    )
+    analytics = Analytics(hass)
+    analytics._data.submission_identifier = "invalid-identifier"
+    await analytics.save_preferences({ATTR_BASE: True, ATTR_SNAPSHOTS: True})
+
+    await analytics.send_snapshot()
+
+    assert analytics._data.submission_identifier is None
+    assert "Invalid submission identifier" in caplog.text
+
+
+async def test_send_snapshot_bad_request(
```

Comment:
this test and `test_send_snapshot_service_unavailable`, `test_send_snapshot_unexpected_status`, `test_send_snapshot_timeout`, `test_send_snapshot_client_error` could be merged by parameterizing the `post` args and expected log